echallan-services/src/main/java/org/egov/echallan/repository/rowmapper/ChallanRowMapper.java

```diff
@@ -0,0 +1,119 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+
+
+
+@Component
+public class ChallanRowMapper  implements ResultSetExtractor<List<Challan>> {
+	@Autowired
+    private ObjectMapper mapper;
+
+    public List<Challan> extractData(ResultSet rs) throws SQLException, DataAccessException {
+        Map<String, Challan> challanMap = new LinkedHashMap<>();
+       
+        while (rs.next()) {
+            String id = rs.getString("challan_id");
+            Challan currentChallan = challanMap.get(id);
+
+            if(currentChallan == null){
+                Long lastModifiedTime = rs.getLong("challan_lastModifiedTime");
+                if(rs.wasNull()){lastModifiedTime = null;}
+
+                Long taxPeriodFrom = (Long) rs.getObject("taxperiodfrom");
+                Long taxPeriodto = (Long) rs.getObject("taxperiodto");
+                PGobject pgObj = (PGobject) rs.getObject("additionaldetail");
+                AuditDetails auditdetails = AuditDetails.builder()
+                        .createdBy(rs.getString("challan_createdBy"))
+                        .createdTime(rs.getLong("challan_createdTime"))
+                        .lastModifiedBy(rs.getString("challan_lastModifiedBy"))
+                        .lastModifiedTime(lastModifiedTime)
+                        .build();
+                try {
+                currentChallan = Challan.builder().auditDetails(auditdetails)
+                		.accountId(rs.getString("uuid"))
+                		.challanNo(rs.getString("challanno"))
+                		.businessService(rs.getString("businessservice"))
+                		.tenantId(rs.getString("tenantid"))
+                		.referenceId(rs.getString("referenceid"))
+                		.taxPeriodFrom(taxPeriodFrom)
+                		.taxPeriodTo(taxPeriodto)
+                		.description(rs.getString("description"))
+                		.applicationStatus(StatusEnum.valueOf(rs.getString("applicationstatus")))
+                		.filestoreid(rs.getString("filestoreid"))
+                        .id(id)
+                        .build();
+                if(pgObj!=null){
+                    JsonNode additionalDetail = mapper.readTree(pgObj.getValue());
+                    currentChallan.setAdditionalDetail(additionalDetail);
+                }
+                }
+                catch (IOException e){
+                    throw new CustomException("PARSING ERROR","Error while parsing additionalDetail json");
+                }
+                challanMap.put(id,currentChallan);
+            }
+            addChildrenToProperty(rs, currentChallan);
+
+        }
+       
+        return new ArrayList<>(challanMap.values());
+
+    }
+
+
+
+    private void addChildrenToProperty(ResultSet rs, Challan challan) throws SQLException {
```

Comment:
Rename the method with the respective object name not property

Reply:
Changes commited